test/fw/ptl/utils/pbs_dshutils.py

```diff
@@ -1946,30 +1947,43 @@ def mkdtemp(self, hostname=None, suffix='', prefix='PtlPbs', dir=None,
                      directory
         :param level: logging level, defaults to INFOCLI2
         """
+        # create a temp dir as current user
+        tmpdir = tempfile.mkdtemp(suffix, prefix, dir)
+
+        # if temp dir to be created on remote host
         if not self.is_localhost(hostname):
-            tmp_args = []
-            if suffix:
-                tmp_args += ['suffix=\'' + suffix + '\'']
-            if prefix:
-                tmp_args += ['prefix=\'' + prefix + '\'']
-            if dir is not None:
-                tmp_args += ['dir=\'' + str(dir) + '\'']
-            args = ",".join(tmp_args)
-            ret = self.run_cmd(hostname,
-                               ['python', '-c', '"import tempfile; ' +
-                                'print tempfile.mkdtemp(' + args + ')"'],
-                               level=level)
-            if ret['rc'] == 0 and ret['out']:
-                fn = ret['out'][0]
-        else:
-            fn = tempfile.mkdtemp(suffix, prefix, dir)
-        if mode is not None:
-            self.chmod(hostname, fn, mode=mode, recursive=True, level=level,
-                       sudo=True)
-        if ((uid is not None) or (gid is not None)):
-            self.chown(hostname, fn, uid=uid, gid=gid, recursive=True,
-                       sudo=True)
-        return fn
+            if asuser is not None:
+                # by default mkstemp creates dir with 0600 permission
+                # to create dir as different user first change the dir
+                # permission to 0644 so that other user has read permission
+                self.chmod(hostname, tmpdir, mode=0644)
```

Comment:
Why are you not including the executable bit?  Shouldn't this be 755? (and below)